### PR TITLE
Feat/gate card options

### PR DESCRIPTION
**Pull request by fbenfraj**
# [PR Name]
gate card options

## Type of PR
- [] - feature

## Description
DAO Profile (as Admin) > Credentials tab > Gate Cards

### CODE REVIEW

File: apps/website/components/molecules/gates-card.tsx [hunk 1]
```diff
@@ -29,28 +37,117 @@ export function GatesCard({
...
+    gqlAuthMethods.deleteGate
+  );
+
+  const deleteGate = () =>
+    deleteGateMutation(
+      { gate_id: id },
+      {
+        onSuccess() {
+          snackbar.onOpen({
+            message: 'Credential deleted!',
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:124]: put content on locales

File: apps/website/components/molecules/gates-card.tsx [hunk 2]
```diff
@@ -29,28 +37,117 @@ export function GatesCard({
...
+      },
+      {
+        onSuccess: async (data) => {
+          setPublished(data.update_gates_by_pk.published);
+
+          snackbar.onOpen({
+            message: `Credential ${
+              published === 'not_published' || published === 'paused'
+                ? 'published!'
+                : 'unpublished!'
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:98]: put content on locales

File: apps/website/components/molecules/gates-card.tsx [hunk 3]
```diff
@@ -29,28 +37,117 @@ export function GatesCard({
...
+      action: () => setConfirmToggleState(true),
+      hidden: false,
+    },
+    {
+      text: 'Edit',
+      action: () => router.push(`${ROUTES.GATE_NEW}?dao=${dao.id}&gate=${id}`),
+      hidden: published === 'published' || published === 'paused',
+    },
+    {
+      text: 'Delete',
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:73]: put content on locales

File: apps/website/components/molecules/gates-card.tsx [hunk 4]
```diff
@@ -29,28 +37,117 @@ export function GatesCard({
...
+    {
+      text:
+        published === 'not_published' || published === 'paused'
+          ? 'Publish'
+          : 'Unpublish',
+      action: () => setConfirmToggleState(true),
+      hidden: false,
+    },
+    {
+      text: 'Edit',
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:68]: put content on locales

File: apps/website/components/molecules/gates-card.tsx [hunk 5]
```diff
@@ -29,28 +37,117 @@ export function GatesCard({
...
+  const [published, setPublished] = useState(publishedProps);
+  const [confirmDelete, setConfirmDelete] = useState(false);
+  const [confirmToggleState, setConfirmToggleState] = useState(false);
+
+  const gateOptions = [
+    {
+      text:
+        published === 'not_published' || published === 'paused'
+          ? 'Publish'
+          : 'Unpublish',
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:63]: put content on locales

File: apps/website/components/molecules/gates-card.tsx [hunk 6]
```diff
@@ -108,18 +213,62 @@ export function GatesCard({
...
+      <ConfirmDialog
+        title="Are you sure you want to delete this credential?"
+        open={confirmDelete}
+        positiveAnswer="Delete"
+        negativeAnswer="Cancel"
+        setOpen={setConfirmDelete}
+        onConfirm={deleteGate}
+      >
+        If you delete this credential, you will not be able to access it and
+        this action cannot be undone.
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:252]: put content on locales

File: apps/website/components/molecules/gates-card.tsx [hunk 7]
```diff
@@ -108,18 +213,62 @@ export function GatesCard({
...
+        anchorOrigin={{
+          vertical: snackbar.vertical,
+          horizontal: snackbar.horizontal,
+        }}
+        open={snackbar.open}
+        onClose={snackbar.handleClose}
+        message={snackbar.message}
+      />
+      <ConfirmDialog
+        title="Are you sure you want to delete this credential?"
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:244]: put content on locales

File: apps/website/components/molecules/gates-card.tsx [hunk 8]
```diff
@@ -108,18 +213,62 @@ export function GatesCard({
...
+        onConfirm={deleteGate}
+      >
+        If you delete this credential, you will not be able to access it and
+        this action cannot be undone.
+      </ConfirmDialog>
+      <ConfirmDialog
+        title={
+          published === 'published'
+            ? 'Are you sure to unpublish this credential?'
+            : 'Are you sure you want to publish this credential?'
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:258]: put content on locales

File: apps/website/components/molecules/gates-card.tsx [hunk 9]
```diff
@@ -108,18 +213,62 @@ export function GatesCard({
...
+        }
+        open={confirmToggleState}
+        positiveAnswer={`${
+          published === 'published' ? 'Unpublish' : 'Publish'
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:262]: put content on locales

File: apps/website/components/molecules/gates-card.tsx [hunk 10]
```diff
@@ -108,18 +213,62 @@ export function GatesCard({
...
+        }`}
+        negativeAnswer="Cancel"
+        setOpen={setConfirmToggleState}
+        onConfirm={toggleGateState}
+      >
+        {published === 'published'
+          ? 'If you unpublish this credential, users will not be able to see it anymore.'
+          : 'Publishing this credential will make it accessible by all users.'}
```
Comment (kbooz) [inline apps/website/components/molecules/gates-card.tsx:270]: put content on locales

**Review by kbooz — CHANGES_REQUESTED** [review]

File: apps/website/components/molecules/gates-card.tsx (hunk 1 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

File: apps/website/components/molecules/gates-card.tsx (hunk 2 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

File: apps/website/components/molecules/gates-card.tsx (hunk 3 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

File: apps/website/components/molecules/gates-card.tsx (hunk 4 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

File: apps/website/components/molecules/gates-card.tsx (hunk 5 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

File: apps/website/components/molecules/gates-card.tsx (hunk 6 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

File: apps/website/components/molecules/gates-card.tsx (hunk 7 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

File: apps/website/components/molecules/gates-card.tsx (hunk 8 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

File: apps/website/components/molecules/gates-card.tsx (hunk 9 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

File: apps/website/components/molecules/gates-card.tsx (hunk 10 above)
Comment (fbenfraj) [reply apps/website/components/molecules/gates-card.tsx]: Done

**Review by kbooz — APPROVED** [review]

**Pull request merged**